lib/embed/media_tag.rb

```diff
@@ -19,22 +19,20 @@ def initialize(ng_doc, viewer)
     attr_reader :ng_document, :purl_document, :request, :viewer
 
     def build_markup
-      file_index = 0
       purl_document.contents.each do |resource|
-        label = resource.description
         next unless SUPPORTED_MEDIA_TYPES.include?(resource.type.to_sym)
+        label = resource.description
         resource.files.each do |file|
           label = file.title if label.blank?
           ng_document.send(
             resource.type.to_sym,
             'data-file-label': label,
             'data-slider-object': file_index,
+            'data-src': streaming_url_for(file, :dash),
```

Comment:
It makes sense that it hasn't all settled down yet.   I was mostly aware that I personally wasn't getting why different info was in different places and how the code was choosing hls vs dash.

Putting all the info in one place could be an improvement.   Dunno if leaving it as is with comments would be viable as well.